examples/custom-server/next.config.js

```diff
@@ -0,0 +1,13 @@
+require('dotenv').config()
+
+module.exports = {
+  publicRuntimeConfig: {
+    SERVER_URL: process.env.PAYLOAD_PUBLIC_SERVER_URL,
+  },
+  images: {
+    domains: [
+      'localhost',
+      // Your domain(s) here
+    ],
+  },
```

Comment:
Based on [suggestion](https://github.com/payloadcms/nextjs-custom-server/issues/23) that works!
```suggestion
  },
  experimental: {
    serverComponentsExternalPackages: ['payload', 'mongoose']
  },
```

Reply:
Isn't this specifically for using the Payload serverlessly? This repo demonstrates server-side fetches, but not Local API usage. If that's the case, this change would likely be more relevant in https://github.com/payloadcms/next-payload

Reply:
Oh, I don't know actually!

Reply:
Ok this likely _would_ be needed for anyone wishing to use `payload` within a server component for any reason. But I think the right move here would be to get `payload` added to this official list: https://github.com/vercel/next.js/blob/canary/packages/next/src/lib/server-external-packages.json

I've opened a PR for this here: https://github.com/vercel/next.js/pull/51933

Reply:
Annnnd it's merged! Once that commit makes its way into a release this change will no longer be needed ✨